MeshKernelNET/test/MeshKernelNETCoreTest/Api/MeshKernelCurvilinearTest.cs

```diff
@@ -0,0 +1,884 @@
+                    curvilinearParameters.SmoothingParameter = 0.5;
+                    curvilinearParameters.AttractionParameter = 0.0;
+                    Assert.IsTrue(api.CurvilinearComputeTransfiniteFromSplines(id, geometryListIn, curvilinearParameters));
+
+                    var curvilinearGrid = new DisposableCurvilinearGrid();
+                    var success = api.CurvilinearGridGetData(id, out curvilinearGrid);
+                    Assert.IsTrue(success);
+                    curvilinearGrid.Dispose();
+                    geometryListIn.Dispose();
+
+                }
+                finally
+                {
+                    api.DeallocateState(id);
+                }
+            }
+        }
+
+
+
+        [Test]
+        public void CurvilinearComputeOrthogonalGridFromSplinesThroughAPI()
+        {
+            // Setup
+            using (var api = new MeshKernelApi())
+            using (var geometryListIn = new DisposableGeometryList())
+            {
+                var id = 0;
+                try
+                {
+                    id = api.AllocateState(0);
+
+                    ;
+                    var geometrySeparator = api.GetSeparator();
+                    geometryListIn.GeometrySeparator = geometrySeparator;
+                    geometryListIn.NumberOfCoordinates = 6;
+
+                    geometryListIn.XCoordinates = new[]
+                    {
+                            1.175014E+02, 3.755030E+02, 7.730054E+02, geometrySeparator,
+                            4.100089E+01, 3.410027E+02
+                        };
+
+                    geometryListIn.YCoordinates = new[]
+                    {
+                            2.437587E+01, 3.266289E+02, 4.563802E+02, geometrySeparator,
+                            2.388780E+02, 2.137584E+01
+                        };
+
+                    geometryListIn.Values = new[]
+                    {
+                            0.0, 0.0, 0.0, geometrySeparator,
+                            0.0, 0.0, geometrySeparator
+                        };
+
+                    var curvilinearParameters = new CurvilinearParameters();
+                    curvilinearParameters.MRefinement = 40;
+                    curvilinearParameters.NRefinement = 10;
+                    var splinesToCurvilinearParameters = SplinesToCurvilinearParameters.CreateDefault();
+                    splinesToCurvilinearParameters.GrowGridOutside = false;
+
+
+                    Assert.IsTrue(api.CurvilinearComputeOrthogonalGridFromSplines(id, geometryListIn,
+                         curvilinearParameters, splinesToCurvilinearParameters));
+
+                    var curvilinearGrid = new DisposableCurvilinearGrid();
+                    var success = api.CurvilinearGridGetData(id, out curvilinearGrid);
+                    Assert.IsTrue(success);
+
+                    curvilinearGrid.Dispose();
+                    geometryListIn.Dispose();
+                }
```

Comment:
Move the Dispose calls to the finally clause. If one of the Assertions fail this will not be called.
Please do this for all subsequent occurrences in this.